plugins/fields/Association/helpers.ts

```diff
@@ -19,11 +19,11 @@ export const getForeignName = (record: any, column: Column) => {
       if (record.title) prettyName = record.title;
       if (record.name) prettyName = record.name;
     }
-  }
 
-  if (prettyName) return `${prettyName} [${record.id}]`;
+    if (prettyName) return `${prettyName} [${record.id}]`;
```

Comment:
These were outside of the `if(record)` conditional. They would have failed on `record.id`. This was the third bug.